scripts/lib/api/generateApiComponents.ts

```diff
@@ -28,6 +32,225 @@ export type ComponentProps = {
   extraRawSignatures?: string[];
 };
 
+const APITYPE_TO_TAG: Record<string, string> = {
+  class: "class",
+  exception: "class",
+  attribute: "attribute",
+  property: "attribute",
+  function: "function",
+  method: "function",
+};
+
+export async function processMdxComponent(
+  $: CheerioAPI,
+  $main: Cheerio<any>,
+  signatures: Cheerio<Element>[],
+  $dl: Cheerio<any>,
+  priorApiType: ApiType | undefined,
+  apiType: ApiType,
+  id: string,
+): Promise<[string, string]> {
+  const $firstSignature = signatures.shift()!;
+
+  const componentProps = prepareProps(
+    $,
+    $main,
+    $firstSignature,
+    $dl,
+    priorApiType,
+    apiType,
+    id,
+  );
+
+  const extraProps = signatures.flatMap(
+    ($overloadedSignature) =>
+      prepareProps($, $main, $overloadedSignature, $dl, apiType, apiType, id) ??
+      [],
+  );
+
+  const tagName = APITYPE_TO_TAG[apiType];
+  addExtraSignatures(componentProps, extraProps);
```

Comment:
Nit: I recommend moving `addExtraSignatures` to live right next to `extraProps` because those two things are related. `tagName` can be moved to the top of the function because it's independent of the rest.